nemo/collections/asr/parts/jasper.py

```diff
@@ -178,19 +178,34 @@ def __init__(
             )
         self.use_mask = use_mask
         self.heads = heads
+        self.same_padding = (self.conv.stride[0] == 1) and (
+            2 * self.conv.padding[0] == self.conv.dilation[0] * (self.conv.kernel_size[0] - 1)
+        )
+
+        # `self.lens` caches consecutive integers from 0 to `self.max_len` that are used to compute the mask for a
+        # batch. Recomputed to bigger size as needed. Stored on a device of the latest batch lens.
+        if self.use_mask:
+            self.max_len = 0
+            self.lens = None
 
     def get_seq_len(self, lens):
+        if self.same_padding:
+            return lens
+
         return (
             lens + 2 * self.conv.padding[0] - self.conv.dilation[0] * (self.conv.kernel_size[0] - 1) - 1
         ) // self.conv.stride[0] + 1
 
     def forward(self, x, lens):
         if self.use_mask:
-            lens = lens.to(dtype=torch.long)
             max_len = x.size(2)
-            mask = torch.arange(max_len).to(lens.device).expand(len(lens), max_len) >= lens.unsqueeze(1)
-            x = x.masked_fill(mask.unsqueeze(1).to(device=x.device), 0)
-            # del mask
+            if max_len > self.max_len:
+                self.lens = torch.arange(max_len)
+                self.max_len = max_len
+
+            self.lens = self.lens.to(lens.device)
```

Comment:
Why not move this instead the `if` statement?

Reply:
`lens` input can come from different device during inference. This will cause mismatch if the inference utterance is shorter than `self.max_len`.